src/main/reactjs/src/hooks/navigation/useBlocker.ts

```diff
@@ -0,0 +1,26 @@
+import { Blocker, History, Transition } from 'history';
+import { useContext, useEffect } from 'react';
+import { UNSAFE_NavigationContext } from 'react-router';
+
+export const useBlocker = (blocker: Blocker, when: boolean) => {
+  const navigator = useContext(UNSAFE_NavigationContext)
+    .navigator as unknown as History;
+
+  useEffect(() => {
+    if (when) {
+      const unblock = navigator.block((tx: Transition) => {
+        const autoUnblockingTx = {
+          ...tx,
+          retry() {
+            unblock();
+            tx.retry();
+          },
+        };
```

Comment:
Mä näen tässä vaan loputtoman rekursion, jossa määriteltäisiin `tx`:lle metodi `retry()`, joka kutsuu itseään jatkuvasti sisäisen määrittämänsä metodin `unblock()` kautta. Oon tosin varmasti ymmärtänyt väärin, mutta ylipäätään tämä `Blocker` alla oleva koodi itselle varsin epäselvä.

Reply:
Eipä tuo itsellenikään täysin selvä ole, mutta näin tää toimii. https://github.com/remix-run/history/blob/main/docs/blocking-transitions.md